feat: Billing page redesign plus credits

sean-brydon opened the pull request:
## What does this PR do?
PR Updates to the new designs for billing + credits

#### Image Demo (if applicable):

<img width="829" height="521" alt="CleanShot 2025-09-18 at 10 42 41" src="https://github.com/user-attachments/assets/7ab2825e-6bd8-440a-82ab-eeac0a2fd095" />

sean-brydon committed:
manage billing section

sean-brydon committed:
wip billing credits

sean-brydon committed:
WIP

sean-brydon committed:
WIP

sean-brydon committed:
Download expense log

sean-brydon committed:
credit worth

sean-brydon committed:
skeleton fixes

sean-brydon committed:
add org tip

sean-brydon committed:
add teams tip

sean-brydon committed:
restore service

sean-brydon committed:
type check

sean-brydon committed:
type check

sean-brydon committed:
fix types

sean-brydon committed:
additional credits

sean-brydon committed:
fix progress bar

sean-brydon committed:
add dashed prop

sean-brydon committed:
match new designs

sean-brydon committed:
Merge branch 'main' into feat/billing-redesign-settings

sean-brydon committed:
Merge branch 'main' into feat/billing-redesign-settings

sean-brydon committed:
hide area with no monthly credits

sean-brydon committed:
fix i18n

sean-brydon committed:
show current balance label

CarinaWolli commented on apps/web/modules/settings/billing/components/BillingCredits.tsx:
```diff
@@ -132,117 +169,154 @@ export default function BillingCredits() {
+                    <p className="text-subtle text-sm font-medium leading-tight">
+                      {orgSlug ? t("credits_per_tip_org") : t("credits_per_tip_teams")}
+                    </p>
+                    <Button
```
I think this should open the dialog and not redirect to `/members`
<img width="623" height="438" alt="Screenshot 2025-09-19 at 10 50 36 AM" src="https://github.com/user-attachments/assets/46cacc38-250a-4983-ad83-9943e868eb7b" />

CarinaWolli commented on apps/web/modules/settings/billing/components/BillingCredits.tsx:
```diff
@@ -132,117 +169,154 @@ export default function BillingCredits() {
+                  <Label>{t("additional_credits")}</Label>
+                  <div className="flex items-center gap-1">
+                    <p className="text-sm font-semibold leading-none">
+                      <span className="text-subtle font-medium">{t("current_balance")}</span>{" "}
```
Current balance should have a little more space below same as Additional Credits
<img width="647" height="72" alt="Screenshot 2025-09-19 at 10 55 27 AM" src="https://github.com/user-attachments/assets/1f2498de-c277-48f7-85d5-44c94dd30eaa" />

CarinaWolli commented on apps/web/modules/settings/billing/components/BillingCredits.tsx:
```diff
@@ -132,117 +169,154 @@ export default function BillingCredits() {
+                  <CreditRow label={t("credits_used")} value={totalUsed} underline="solid" />
+                  <CreditRow
+                    label={t("total_credits_remaining")}
+                    value={creditsData.credits.totalRemainingCreditsForMonth}
```
This should only be monthly credits not monthly + additional

<img width="857" height="250" alt="Screenshot 2025-09-19 at 11 02 06 AM" src="https://github.com/user-attachments/assets/ab0fcf66-5793-4f37-a2d4-18440c4a05b2" />

sean-brydon committed:
Update apps/web/modules/settings/billing/billing-view.tsx

Co-authored-by: Carina Wollendorfer <30310907+CarinaWolli@users.noreply.github.com>

sean-brydon committed:
spacing + monthly credits not showing additional

sean-brydon committed:
Remove additional credits from monthly calculations

devin-ai-integration[bot] committed:
feat: replace add members redirect with invite modal in billing settings

- Replace Button href with onClick handler to open MemberInvitationModal
- Add MemberInvitationModalWithoutMembers import and state management
- Maintain existing team/org context support
- Follow established modal usage patterns from other components
- Fix lint error by using undefined instead of empty arrow function

Co-Authored-By: sean@cal.com <Sean@brydon.io>

sean-brydon committed:
Remove redudant vars from method

sean-brydon committed:
fix type check

anikdhabal committed:
Merge branch 'main' into feat/billing-redesign-settings
